src/python/pants/backend/codegen/tasks/protobuf_gen.py

```diff
@@ -136,18 +137,25 @@ def execute_codegen(self, target, target_workdir):
       protoc_environ['PATH'] = os.pathsep.join(self._extra_paths
                                                + protoc_environ['PATH'].split(os.pathsep))
 
+    # Note: The test_source_ordering integration test scrapes this output, so modify it with care.
     self.context.log.debug('Executing: {0}'.format('\\\n  '.join(args)))
-    process = subprocess.Popen(args, env=protoc_environ)
-    result = process.wait()
-    if result != 0:
-      raise TaskError('{0} ... exited non-zero ({1})'.format(self.protobuf_binary, result))
+    with self.context.new_workunit(name='protoc',
+                                   labels=[WorkUnitLabel.TOOL],
+                                   cmd=' '.join(args)) as workunit:
+      result = subprocess.call(args,
+                               env=protoc_environ,
+                               stdout=workunit.output('stdout'),
+                               stderr=workunit.output('stderr'))
+      if result != 0:
+        raise TaskError('{} ... exited non-zero ({})'.format(self.protobuf_binary, result))
```

Comment:
While you are at it, do you think it would be more user friendly to show the command as well?  e.g. https://github.com/wisechengyi/pants/blob/1e45167fab9a30f32207d025d91d2a071c8981fe/src/python/pants/backend/graph_info/tasks/cloc.py#L71-L71?